packages/replay-next/src/contexts/FocusContext.tsx

```diff
@@ -94,33 +100,43 @@ export function FocusContextRoot({ children }: PropsWithChildren<{}>) {
   }, [loadedRegions]);
 
   const updateFocusRange = useCallback(
-    async (value: Range | null, debounce: boolean) => {
-      let newRange: TimeStampedPointRange | null = null;
-      if (value) {
-        const [timeBegin, timeEnd] = value;
-        const pointBegin = await imperativelyGetClosestPointForTime(client, timeBegin);
-        const pointEnd = await imperativelyGetClosestPointForTime(client, timeEnd);
-
-        newRange = {
-          begin: { point: pointBegin, time: timeBegin },
-          end: { point: pointEnd, time: timeEnd },
-        };
-      }
-
-      setRange(newRange);
+    async (range: TimeStampedPointRange | null, debounce: boolean) => {
+      setRange(range);
 
       // Focus values may change rapidly (e.g. during a click-and-drag)
       // In this case, React's default high/low priority split is helpful, but we can do more.
       // Debouncing a little before starting the transition helps avoid sending a lot of unused requests to the backend.
       if (debounce) {
-        debouncedSetDeferredRange(newRange);
+        debouncedSetDeferredRange(range);
       } else {
         startTransition(() => {
-          setDeferredRange(newRange);
+          setDeferredRange(range);
         });
       }
     },
-    [client, debouncedSetDeferredRange]
+    [debouncedSetDeferredRange]
+  );
+
+  const updateForTimelineImprecise = useCallback(
+    async (value: TimeRange | null, debounce: boolean) => {
+      if (value) {
+        const [timeBegin, timeEnd] = value;
+
+        const pointBegin = await imperativelyGetClosestPointForTime(client, timeBegin);
```

Comment:
:question: Tiny nit/suggestion: can we `Promise.all()` these?

Reply:
Probably not. This portion was just a copy-paste but I don't mind making that change.